Added disclaimer on offsetting with new teststatistics

Zeff020 opened the pull request:
# This Pull request:

## Changes or fixes:

Adds some info on the special case where a user tries to make a new-style teststatistics based likelihood and requests to minimise it in serial mode with offsetting enabled. This case is currently not yet supported. The added comment in `RooMinimizer.cxx` explains in more detail why this does not work currently, and gives a suggestion for how to make this work in the future.

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

guitargeek commented:
@phsft-bot build

guitargeek commented:
Hi, thanks for the PR! And nice, it seems you have uncovered a unit test that did not really work with the offsetting to begin with!

How do you want to proceed? Can the unit test be fixed, or does it need to be disabled?

guitargeek commented:
Hi @Zeff020 and @egpbos!

Can this PR be closed without merging after #11717 is merged, or is the disclaimer still relevant then?

Thanks!

Zeff020 commented:
Hi @guitargeek,

This disclaimer is still relevant after that MR that you referred to is merged. This MR is about the RooRealL passed to the RooMinimizerFcn not having offsetting implemented, and really not even being intended for minimisation. The MR that you linked to does not address this issue as far as I can tell.

guitargeek commented:
Alright, thanks for the clarification! Then you need to rebase this PR now and force-push to the `roofit-dev/offsetting-teststatistics-interface` Maybe we can get around having to open a new PR just for the rebasing this time, to not spam the list of closed PRs to ROOT too much.

If I remember correctly, the problem was that `roofit-dev` didn't allow force pushes? Maybe this can be changed? Thanks for considering this!

egpbos commented:
Probably best to wait for #11717, since there I also edited RooMinimizer, so we'll again have a conflict after that.

Zeff020 commented:
Hi @guitargeek, I have rebased and force-pushed this branch as you requested.

guitargeek commented:
Thank you, excellent!

guitargeek commented:
@phsft-bot  build

Zeff020 commented:
Hi @guitargeek, I fixed the tests which were now (correctly) failing basically by the suggestions given by @egpbos here before and also added a new test for specifically this behaviour. After a successful build and test run this should be fine to merge from my side now.

egpbos commented:
@phsft-bot build

guitargeek commented on roofit/roofitcore/src/RooMinimizer.cxx:
```diff
@@ -124,8 +124,16 @@ RooMinimizer::RooMinimizer(RooAbsReal &function, Config const &cfg) : _cfg(cfg)
+         // minimization altogether.
          _fcn = std::make_unique<RooMinimizerFcn>(&function, this);
-      }
+      } 
```
```suggestion
      }
```
Please don't introduce this additional whitespace

guitargeek commented:
Hi @Zeff020! Sorry for asking this again, but after https://github.com/root-project/root/pull/11793 was merged you got to rebase and force push again to resolve another conflict. Also, please fix this trailing whitespace that is introduced by this PR when you are already at it.

Sorry again for this, your PR will be merged next!

Zeff020 commented:
Changes implemented @guitargeek

guitargeek commented:
@phsft-bot build

egpbos commented:
CI passed 🎉

guitargeek reviewed (approved):
LGTM!
